### ISSUE TITLE
Improove handeling of pre-conditions for installation

### ISSUE THREAD

**Issue by DiogoConstantino** [ubports/ubports-installer#1599]
The installer does suggest the user to verify certain pre-conditions for installing the OS. However I do believe it could provide even more useful information or even allow the user to authorize the Installer to make necessary changes on the device to enforce those pre-conditions (ideally also to rollback if it failed, doing those changes, or installing).

Examples:
- When installing on Nexus 5, the installer warns the user about the possible need for re-partiion the device, and also points the user to a forum thread where he can learn more about it. This is good, but it should go even further, it should check if that is really needed on that specific device, and if it's need offer to do it for the user.

- When installing on the BQ M10 family of devices, the installer warns the user about the need to select the correct image (for frieza, or cooler), whoever it doesn't offer information to the user on how the use can identify the model correctly. I suspect it's not possible for the Installer to diferenciate between those tow, and that's why it asks the user to do so, but if that's the case, I also believe it can be even more helpful and provide an authoritative link with instructions.

**Comment by NeoTheThird** [comment]
Yeah, the pre-conditions are being over-used a little right now. They're a work-around for things we can't automate yet, but i agree that we should try and get better at automating and detecting more stuff. Gonna be challenging, though.

For distinguishing the models, config structure `v2` provides the new `doppelgangers` field, which will be used to implement a proper UI for this. Maybe you're right, and we should also list identification features, but we have to be careful not to overload the user with information. For cases like the M10 HD and FHD, i think most normal users will know which one they have. Let's hope we don't get too many devices where the same marketing name applies to vastly different chipsets... We'll see.
